### PR TITLE
finished assessment

### PR DESCRIPTION
**Pull request by gmgreer**
# Submitting Week 7 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -43,4 +82,23 @@ const cubeAndSum1 = [2, 3, 4]
...
+})
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+
 // b) Create the function that makes the test pass.
+// create a function that takes an array of numbers as input
+sumCubed = (arr) => {
+  // use the reduce higher order function. Reduce takes an array and returns a single value. Two arguments the first is the accumulated value, the second is the input from the loop. Cube each number. The 0 after the comma is the initial value.
+  return arr.reduce((sum, num) => num ** 3 + sum, 0)
```
Comment (elysemontano) [inline code-challenges.test.js:103]: Outstanding work in this section!!!

File: interview-questions.md [hunk 2]
```diff
@@ -8,36 +8,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+add_foreign_key :uploads, :users
+then of course rails db:migrate 
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: Lets go through, so from what I understand the Get command will need to know what it is getting and if it is getting a single item will need the primary key for that item. I think any get even if you are getting all will have to pass the table name in params. Or maybe that can be hard coded so you wouldn't need the params. For a Post or Create you will need the information you are posting so that will need params. For both of the updates you will need the information to be updated as well as the id of that think in params. For destroy you will need the id of the thing you want to destroy. So all but maybe the get all will need params. 
 
-Researched answer:
+Researched answer: So Index will not need params though I am sure you could set it up to where it has params. But it doesn't need it because it is just doing a .all. Then there is new which doesn't need params since it is just passing the information from the user and packaging it up in a format that will later be passed as params.
```
Comment (elysemontano) [inline interview-questions.md:22]: This question is kind of tricky, but you are heading in the right direction.  Show, Update, Edit, and Destroy need params passed in the url to be able to know which instance you wish to interact with.

File: rails-commenting.rb [hunk 3]
```diff
@@ -33,12 +38,14 @@ def create
 
   def edit
     # ---6)
+    #find the item that needs to be edited based on the id given. Probably when you hit the update button it takes the current id params that were passed when you selected the item you wanted to update
```
Comment (elysemontano) [inline rails-commenting.rb:41]: New and Edit have the main purpose of setting up a form for us to view.

**Review by elysemontano — APPROVED** [review]
Excellent work as always Garrett!  I left some comments for you to take a look at.
Current assessment level is competent - You continue to show a strong understanding of the concepts.  You 100% crushed the code challenges section with efficient and well written code and tests. To stretch yourself, consider edge cases for those problems as well.  It is always helpful to review RESTful routes as this is going to be foundational for full stack applications.  Keep up the amazing work!
